### PR TITLE
Update REST API documentation for MLflow 1.0

### PR DESCRIPTION
**Pull request by aarondav**
## What changes are proposed in this pull request?
 
Regenerates API docs for MLflow 1.0.
 
### Is this a user-facing change? 

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [x] REST-API 
- [x] Docs
### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/rest-api.rst [hunk 1]
```diff
@@ -1324,78 +1211,6 @@ Tag for a run.
 | value      | ``STRING`` | The tag value. |
 +------------+------------+----------------+
 
-Tag keys that start with ``mlflow.`` are reserved for internal use. The following tags are set
```
Comment (sueann) [inline docs/source/rest-api.rst]: Don't we want to keep this unfortunately-manually-written part?

Comment (aarondav) [reply docs/source/rest-api.rst]: Moved: https://github.com/mlflow/mlflow/blob/master/docs/source/tracking.rst#id95

Comment (sueann) [reply docs/source/rest-api.rst]: nice - thanks!

**Review by sueann — APPROVED** [review]
lgtm!

**Pull request merged**